one_time_sale/models/sale_order.py

```diff
@@ -0,0 +1,35 @@
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.depends('order_line', 'order_line.recurring_invoice')
+    def _compute_has_recurring_line(self):
+        """Disable validation of plan_id for one-time product purchase."""
+        recurring_orders = self.filtered(lambda order: order.order_line.filtered(
+            lambda line: line.product_id.recurring_invoice and not line.product_id.accept_one_time
+        ))
+        recurring_orders.has_recurring_line = True
+        (self - recurring_orders).has_recurring_line = False
```

Comment:
Always try to make code readable unless the code you written is improve the performance here is suggestion 
```suggestion
        for order in self:
            order.has_recurring_line = any(
                line.product_id.recurring_invoice and not line.product_id.accept_one_time
                for line in order.order_line
            )
```
             
its more readable but i haven't tested it could you check it and apply it  accordingly?